### PR TITLE
applications: nrf_desktop: Add nrf52820 dongle

### PR DESCRIPTION
**Pull request by pdunaj**
Jira:DESK-870

### CODE REVIEW

**Comment by pdunaj** [comment]
Dongle works like a charm but we need to get nrf52820 support from the upstream. I guess we could merge it now anyway.

File: boards/arm/nrf52820dongle_nrf52820/Kconfig.defconfig [hunk 1]
```diff
@@ -4,12 +4,12 @@
...
 
 config BOARD
-	default "nrf52833donglev2_nrf52833"
+	default "nrf52820dongle_nrf52820"
 
 config BT_CTLR
 	default BT
 
-endif # BOARD_NRF52833DONGLEV2_NRF52833
+endif # BOARD_NRF52820DONGLEV2_NRF52820
```
Comment (MarekPieta) [inline boards/arm/nrf52820dongle_nrf52820/Kconfig.defconfig:15]: Fix the comment

File: boards/arm/nrf52820dongle_nrf52820/nrf52820dongle_nrf52820.dts [hunk 2]
```diff
@@ -5,13 +5,13 @@
...
+#include <nordic/nrf52820_qdaa.dtsi>
 
 / {
-	model = "Nordic nRF52833 Donglev2 NRF52833";
-	compatible = "nordic,nrf52833-donglev2-nrf52833",
-		     "nordic,nrf52833-qiaa",
-		     "nordic,nrf52833";
+	model = "Nordic nRF52820 Dongle NRF52820";
+	compatible = "nordic,nrf52820-dongle-nrf52820",
+		     "nordic,nrf52820-qiaa",
```
Comment (MarekPieta) [inline boards/arm/nrf52820dongle_nrf52820/nrf52820dongle_nrf52820.dts:13]: "nordic,nrf52820-qdaa"

File: applications/nrf_desktop/configuration/nrf52820dongle_nrf52820/port_state_def.h [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+ */
+const struct {} port_state_def_include_once;
+
+
+static const struct pin_state port0_on[] = {
+	{DT_GPIO_LEDS_LED1_RED_GPIOS_PIN, 1},
+	{DT_GPIO_LEDS_LED1_GREEN_GPIOS_PIN, 1},
+};
+
+static const struct pin_state port1_on[] = {
```
Comment (MarekPieta) [inline applications/nrf_desktop/configuration/nrf52820dongle_nrf52820/port_state_def.h:26]: We do not need the port1 arrays here

File: applications/nrf_desktop/configuration/nrf52820dongle_nrf52820/leds_def.h [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+ * information about LED mapping to PWM channels.
+ */
+
+/* This structure enforces the header file is included only once in the build.
+ * Violating this requirement triggers a multiple definition error at link time.
+ */
+const struct {} leds_def_include_once;
+
+/* Mapping the PWM channels to pin numbers */
+static const size_t led_pins[CONFIG_DESKTOP_LED_COUNT]
```
Comment (MarekPieta) [inline applications/nrf_desktop/configuration/nrf52820dongle_nrf52820/leds_def.h:17]: LEDs are not used, but I think we can leave it - we will add support for them soon

File: boards/arm/nrf52820dongle_nrf52820/Kconfig.defconfig (hunk 1 above)
Comment (pdunaj) [reply boards/arm/nrf52820dongle_nrf52820/Kconfig.defconfig:15]: fixed

File: boards/arm/nrf52820dongle_nrf52820/nrf52820dongle_nrf52820.dts (hunk 2 above)
Comment (pdunaj) [reply boards/arm/nrf52820dongle_nrf52820/nrf52820dongle_nrf52820.dts:13]: Fixed

File: applications/nrf_desktop/configuration/nrf52820dongle_nrf52820/port_state_def.h (hunk 3 above)
Comment (pdunaj) [reply applications/nrf_desktop/configuration/nrf52820dongle_nrf52820/port_state_def.h:26]: No need for it - removed

File: applications/nrf_desktop/configuration/nrf52820dongle_nrf52820/leds_def.h (hunk 4 above)
Comment (pdunaj) [reply applications/nrf_desktop/configuration/nrf52820dongle_nrf52820/leds_def.h:17]: Maybe or maybe not but for sure this won't be using PWM. I will remove this config

**Comment by pdunaj** [comment]
I have fixed the issues found and also added a config with the mcuboot for serial recovery. It's working.

File: boards/arm/nrf52820dongle_nrf52820/CMakeLists.txt [hunk 5]
```diff
@@ -0,0 +1,9 @@
+#
+# Copyright (c) 2020 Nordic Semiconductor ASA
+#
+# SPDX-License-Identifier: LicenseRef-BSD-5-Clause-Nordic
+#
+
+zephyr_library()
+zephyr_library_sources(board.c)
+zephyr_library_include_directories(${ZEPHYR_BASE}/drivers)
```
Comment (anangl) [inline boards/arm/nrf52820dongle_nrf52820/CMakeLists.txt:9]: What do you actually need this line for?

File: boards/arm/nrf52820dongle_nrf52820/board.c [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+		}
+
+		NRF_UICR->REGOUT0 =
+		    (NRF_UICR->REGOUT0 & ~((uint32_t)UICR_REGOUT0_VOUT_Msk)) |
+		    (UICR_REGOUT0_VOUT_3V0 << UICR_REGOUT0_VOUT_Pos);
+
+		NRF_NVMC->CONFIG = NVMC_CONFIG_WEN_Ren << NVMC_CONFIG_WEN_Pos;
+		while (NRF_NVMC->READY == NVMC_READY_READY_Busy) {
+			;
+		}
```
[the remark this answers is not in the document]
Comment (pdunaj) [reply boards/arm/nrf52820dongle_nrf52820/board.c:36]: I copied it from 833 dongle and there from 840 dongle

File: boards/arm/nrf52820dongle_nrf52820/nrf52820dongle_nrf52820.dts [hunk 7]
```diff
@@ -5,13 +5,13 @@
...
 
 / {
-	model = "Nordic nRF52833 Donglev2 NRF52833";
-	compatible = "nordic,nrf52833-donglev2-nrf52833",
-		     "nordic,nrf52833-qiaa",
-		     "nordic,nrf52833";
+	model = "Nordic nRF52820 Dongle NRF52820";
+	compatible = "nordic,nrf52820-dongle-nrf52820",
+		     "nordic,nrf52820-qdaa",
+		     "nordic,nrf52820";
```
[the remark this answers is not in the document]
Comment (pdunaj) [reply boards/arm/nrf52820dongle_nrf52820/nrf52820dongle_nrf52820.dts:14]: can fix it; it is again a mirror of 840 dongle

[the remark this answers is not in the document]
Comment (pdunaj) [reply boards/arm/nrf52820dongle_nrf52820/nrf52820dongle_nrf52820.dts:14]: done

File: boards/arm/nrf52820dongle_nrf52820/CMakeLists.txt (hunk 5 above)
Comment (pdunaj) [reply boards/arm/nrf52820dongle_nrf52820/CMakeLists.txt:9]: another copy paste from the 840 dongle - will remove it unless something complains

Comment (pdunaj) [reply boards/arm/nrf52820dongle_nrf52820/CMakeLists.txt:9]: done

File: boards/arm/nrf52820dongle_nrf52820/board.c (hunk 6 above)
[the remark this answers is not in the document]
Comment (pdunaj) [reply boards/arm/nrf52820dongle_nrf52820/board.c:36]: done

**Comment by pdunaj** [comment]
Hi @anangl , thanks for the review. I have also applied the same cleanup to 833 dongle.

**Pull request merged**